### PR TITLE
feat: Improve DX for self-hosted use-cases

### PR DESCRIPTION
**Pull request by m1guelpf**
Needs input on:

- Final name for IDKit prop (currently `unstable_selfHosted`)
- Final name for self-hosted app_id (currently `self_hosted`)

### CODE REVIEW

File: packages/react/src/types/config.ts [hunk 1]
```diff
@@ -22,7 +22,9 @@ export type WidgetConfig = {
 	onError?: CallbackFn<IErrorState>
 }
 
-export type Config = IDKitConfig & Required<Pick<IDKitConfig, 'action'>> & WidgetConfig
+export type Config = Required<Pick<IDKitConfig, 'action'>> &
+	WidgetConfig &
+	((Exclude<IDKitConfig, 'app_id'> & { unstable_selfHosted: true }) | (IDKitConfig & { unstable_selfHosted?: false }))
```
Comment (pdtfh) [inline packages/react/src/types/config.ts:27]:
```suggestion
	((Exclude<IDKitConfig, 'app_id'> & { advanced_selfHosted: true }) | (IDKitConfig & { advanced_selfHosted?: false }))
```

thoughts?

File: packages/react/src/store/idkit.ts [hunk 2]
```diff
@@ -13,8 +13,10 @@ import {
 	type ISuccessResult,
 } from '@worldcoin/idkit-core'
 
+const SELF_HOSTED_APP_ID = 'self_hosted'
+
 export type IDKitStore = {
-	app_id: IDKitConfig['app_id']
+	app_id: IDKitConfig['app_id'] | typeof SELF_HOSTED_APP_ID | ''
```
Comment (pdtfh) [inline packages/react/src/store/idkit.ts:19]: why can it be an empty string?

**Review by pdtfh — APPROVED** [review]

File: packages/react/src/store/idkit.ts (hunk 2 above)
Comment (0xPenryn) [reply packages/react/src/store/idkit.ts:19]: why do we require an app_id at all when it's self-hosted? why not just force the app_id to `self-hosted` if the `advanced_selfHosted` param is true?

File: packages/react/src/types/config.ts (hunk 1 above)
Comment (m1guelpf) [reply packages/react/src/types/config.ts:27]: feels better to me, would still love to hear @0xPenryn's thoughts on snake_case+camelCase tho

Comment (pdtfh) [reply packages/react/src/types/config.ts:27]: aligned sync, we'll do `advanced.self_hosted`

File: packages/react/src/types/config.ts [hunk 3]
```diff
@@ -20,7 +20,9 @@ export type WidgetConfig = {
 	onError?: CallbackFn<IErrorState>
 }
 
-export type Config = IDKitConfig & Required<Pick<IDKitConfig, 'action'>> & WidgetConfig
+export type Config = Required<Pick<IDKitConfig, 'action'>> &
+	WidgetConfig &
+	((Exclude<IDKitConfig, 'app_id'> & { advanced: { selfhosted: true } }) | (IDKitConfig & { advanced?: { selfhosted?: false } }))
```
Comment (pdtfh) [inline packages/react/src/types/config.ts:25]: shouldn't it be either self_hosted or selfHosted?

**Review by pdtfh — APPROVED** [review]

**Pull request merged**